src/formpack/pack.py

```diff
@@ -202,11 +202,41 @@ def _combine_field_choices(old_field, new_field):
             old_choice = old_field.choice
             new_choice = new_field.choice
         except AttributeError:
-            return
+            return new_field
         combined_options = old_choice.options.copy()
         combined_options.update(new_choice.options)
         new_choice.options = combined_options
 
+        # Copy value_names as well. Even if some options have been deleted,
```

Comment:
I think the work you're doing here to fix up the options could be simplified to what's done by the [`FormChoiceFieldWithMultipleSelect`](https://github.com/kobotoolbox/formpack/blob/d4e4ce0d52a2d2eca24aaaf90a688b80cb88df4a/src/formpack/schema/fields.py#L713-L714) and [`FormDataDef`](https://github.com/kobotoolbox/formpack/blob/d4e4ce0d52a2d2eca24aaaf90a688b80cb88df4a/src/formpack/schema/datadef.py#L27) constructors.

Would you endorse adding a `replace_choice(choice)` method to `FormChoiceFieldWithMultipleSelect` that:

1. Sets `choice` (i.e. what we've created here as `new_choice`) on the field;
1. Does `self.value_names = self.get_value_names()`, duplicating what's done in the `FormDataDef` constructor;
1. Does `self.empty_result = dict.fromkeys(self.empty_result, '')` (from the `FormChoiceFieldWithMultipleSelect` constructor)?